posthog/temporal/llm_analytics/trace_summarization/workflow.py

```diff
@@ -0,0 +1,176 @@
+                generate_and_save_summary_activity,
+                args=[
+                    trace_id,
+                    team_id,
+                    window_start,
+                    window_end,
+                    mode,
+                    batch_run_id,
+                    model,
+                ],
+                schedule_to_close_timeout=timedelta(seconds=GENERATE_SUMMARY_TIMEOUT_SECONDS),
+                retry_policy=constants.SUMMARIZE_RETRY_POLICY,
+            )
+
+    @temporalio.workflow.run
+    async def run(self, inputs: BatchSummarizationInputs) -> BatchSummarizationResult:
+        """
+        Execute batch summarization workflow.
+
+        Args:
+            inputs: BatchSummarizationInputs containing workflow parameters
+
+        Returns:
+            BatchSummarizationResult containing metrics and results
+        """
+        start_time = temporalio.workflow.now()
+        batch_run_id = f"{inputs.team_id}_{start_time.isoformat()}"
+        metrics = BatchSummarizationMetrics()
+
+        # Compute window dates for trace queries
+        if inputs.window_start and inputs.window_end:
+            window_start = inputs.window_start
+            window_end = inputs.window_end
+        else:
+            now = temporalio.workflow.now()
+            window_end = now.isoformat()
+            window_start = (now - timedelta(minutes=inputs.window_minutes)).isoformat()
+
+        # Get trace IDs in window
+        trace_ids = await temporalio.workflow.execute_activity(
+            query_traces_in_window_activity,
+            inputs,
+            schedule_to_close_timeout=timedelta(seconds=SAMPLE_TIMEOUT_SECONDS),
+            retry_policy=constants.SAMPLE_RETRY_POLICY,
+        )
+        metrics.traces_queried = len(trace_ids)
+
+        # Process traces in batches
+        semaphore = asyncio.Semaphore(inputs.batch_size)
+        tasks: list[asyncio.Task[SummarizationActivityResult]] = [
+            self._process_trace(
+                semaphore=semaphore,
+                trace_id=trace_id,
+                team_id=inputs.team_id,
+                window_start=window_start,
+                window_end=window_end,
+                mode=inputs.mode,
+                batch_run_id=batch_run_id,
+                model=inputs.model,
+            )
+            for trace_id in trace_ids
+        ]
+        results: list[SummarizationActivityResult] = await asyncio.gather(*tasks, return_exceptions=False)
+
+        successful_trace_ids = []
+        for result in results:
+            if result.success:
+                successful_trace_ids.append(result.trace_id)
+            elif result.skipped:
+                metrics.summaries_skipped += 1
+
+        metrics.summaries_generated = len(successful_trace_ids)
+        metrics.summaries_failed = metrics.traces_queried - metrics.summaries_generated - metrics.summaries_skipped
+
+        # Add a delay to allow events to be processed
+        await asyncio.sleep(5)
```

Comment:
I think `await temporalio.workflow.sleep(5)` would be more appropriate here

Reply:
https://github.com/PostHog/posthog/pull/41765/commits/ea3fd2726b8d6bc0975e0edf68bf718d15d52a85